### PR TITLE
Adding, retrieving and deleting pipelines in API policies along with test cases

### PR DESCRIPTION
**Pull request by dinusha92**
### Proposed changes in this pull request

- Added Methods to add Pipeline and Conditions in side pipelines to the database
- Added test cases to cover the above mentioned methods 
- Modify SampleTestObjectCreator to support add Pipelines and bandwidth volume based limit when default API creating
- Added test cases(APIGatewayPublisherImplTestCase) to cover a portion in APIGatewayPublisherImpl 
- Removed getSubscriptionPolicybyID() as it was not used anywhere

### When should this PR be merged

- No any precondition

### Follow up actions

- No follow up actions

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] ** Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] ** Are all UI and API inputs run through forms or serializers?**
- [ ] ** Are all external inputs validated and sanitized appropriately?**
- [ ] ** Does all branching logic have a default case?**
- [ ] ** Does this solution handle outliers and edge cases gracefully?**
- [ ] ** Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Comment by pubudu538** [comment]
Shall we avoid adding merge commits? You can rebase the code.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java [hunk 1]
```diff
@@ -94,22 +95,41 @@ public void addPolicy(String policyLevel, Policy policy) throws APIMgtDAOExcepti
             connection = DAOUtil.getConnection();
 
             //TODO : instead of checking policyLevel, check class type, and remove passing policy level to here
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java]: Shall we add all the missing method comments as in https://github.com/wso2/carbon-apimgt/blob/C5/components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/LabelDAOImpl.java#L32

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java [hunk 2]
```diff
@@ -94,22 +95,41 @@ public void addPolicy(String policyLevel, Policy policy) throws APIMgtDAOExcepti
...
+            if (policy instanceof APIPolicy) {
+                String type = policy.getDefaultQuotaPolicy().getType();
+                Limit limit = policy.getDefaultQuotaPolicy().getLimit();
+                if (type.equals(PolicyConstants.REQUEST_COUNT_TYPE)) {
+                    if (limit instanceof RequestCountLimit) {
+                        addAPIPolicy(connection, policy.getPolicyName(), policy.getDisplayName(),
+                                policy.getDescription(),
+                                policy.getDefaultQuotaPolicy().getType(),
+                                10,
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java]: Shall we format the code properly?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java (hunk 1 above)
Comment (dinusha92) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java]: Method comments added

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java (hunk 2 above)
Comment (dinusha92) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java]: Formatted the code

**Pull request closed without merging**